mm/mm_heap/mm_sem.c

```diff
@@ -77,29 +54,27 @@ void mm_seminitialize(FAR struct mm_heap_s *heap)
    */
 
   _SEM_INIT(&heap->mm_semaphore, 0, 1);
-
-  heap->mm_holder      = NO_HOLDER;
-  heap->mm_counts_held = 0;
 }
 
 /****************************************************************************
- * Name: mm_trysemaphore
+ * Name: mm_takesemaphore
  *
  * Description:
- *   Try to take the MM mutex.  This is called only from the OS in certain
+ *   Take the MM mutex.  This may be called from the OS in certain
  *   conditions when it is necessary to have exclusive access to the memory
  *   manager but it is impossible to wait on a semaphore (e.g., the idle
  *   process when it performs its background memory cleanup).
```

Comment:
The description needs to be completely rewritten, it is not adequate for the current implementation.
Also add the documentation for the input parameters and return value.

BTW, the return value documentation may be the most suitable place to describe that scenario where the caller may fail to take the semaphore.

Reply:
Done.